// MessageType
syntax = "proto3";
package protobuf;

enum GameState // 
{
    NULL_GAME_STATE = 0;
    GAME_START = 1;
    GAME_RUNNING = 2;
    GAME_END = 3;
}

enum PlaceType // 
{
    NULL_PLACE_TYPE = 0;
    HOME = 1;              // 出生点（地图左下与右上）
    SPACE = 2;             // 空地
    BARRIER = 3;           // 障碍
    BUSH = 4;              // 草丛
    ECONOMY_RESOURCE = 5;  // 经济资源
    ADDITION_RESOURCE = 6; // 加成资源
    CONSTRUCTION = 7;      // 建筑
    TRAP = 8;              // 陷阱
}

enum ShapeType // 
{
    NULL_SHAPE_TYPE = 0;
    CIRCLE = 1;
    SQUARE = 2;
}

enum CharacterType // 
{
    NULL_CHARACTER_TYPE = 0;
    // 取经团队阵营角色
    TangSeng = 1;
    SunWukong = 2;
    ZhuBajie = 3;
    ShaWujing = 4;
    BaiLongma = 5;
    Monkid = 6;
    
    // 妖怪团队阵营角色
    JiuLing = 7;
    HongHaier = 8;
    NiuMowang = 9;
    TieShan = 10;
    ZhiZhujing = 11;
    Pawn = 12;
}

enum CharacterState // 角色状态
{
    NULL_CHARACTER_STATE = 0;

    IDLE = 1;
    HARVESTING = 2;
    ATTACKING = 3;
    SKILL_CASTING = 4;
    CONSTRUCTING = 5;
    MOVING = 6;


    BLIND = 7;
    KNOCKED_BACK = 8;
    STUNNED = 9;
    INVISIBLE = 10;
    HEALING = 11;
    BERSERK = 12;
    BURNED = 13;
    DECEASED = 14;
}


enum EconomyResourceType // 经济资源
{
    NULL_ECONOMY_RESOURCE_TYPE = 0;

    SMALL_ECONOMY_RESOURCE = 1;
    MEDIUM_ECONOMY_RESOURCE = 2;
    LARGE_ECONOMY_RESOURCE = 3;
}

enum AdditionResourceType // 加成资源
{
    NULL_ADDITION_RESOURCE_TYPE = 0;

    LIFE_POOL1 = 1;
    LIFE_POOL2 = 2;
    LIFE_POOL3 = 3;

    CRAZY_MAN1 = 4;
    CRAZY_MAN2 = 5;
    CRAZY_MAN3 = 6;

    QUICK_STEP = 7;

    WIDE_VIEW = 8;
}

enum EconomyResourceState // 
{
    NULL_ECONOMY_RESOURCE_STSTE = 0;
    HARVESTABLE = 1;
    BEING_HARVESTED = 2;
    HARVESTED = 3;
}

enum AdditionResourceState // 
{
    NULL_ADDITION_RESOURCE_STATE = 0;
    BEATABLE = 1;
    BEING_BEATEN = 2;
    BEATEN = 3;
}

enum EquipmentType // 装备
{
    NULL_EQUIPMENT_TYPE = 0;

    SMALL_HEALTH_POTION = 1;
    MEDIUM_HEALTH_POTION = 2;
    LARGE_HEALTH_POTION = 3;

    SMALL_SHIELD = 4;
    MEDIUM_SHIELD = 5;
    LARGE_SHIELD = 6;
    
    SPEEDBOOTS = 7;
    PURIFICATION_POTION = 8;
    INVISIBILITY_POTION = 9;
    BERSERK_POTION = 10;
}

enum ConstructionType // 建筑
{
    NULL_CONSTRUCTION_TYPE = 0;
    BARRACKS = 1;
    SPRING = 2;
    FARM = 3;
}

enum TrapType // 陷阱
{
    NULL_TRAP_TYPE = 0;
    HOLE = 1; // 坑洞陷阱
    CAGE = 2; // 牢笼陷阱
}

enum NewsType // 
{
    NULL_NEWS_TYPE = 0;
    TEXT = 1;
    BINARY = 2;
}

enum PlayerTeam // 
{
    NULL_TEAM = 0;
    BUDDHISTS_TEAM=1;
    MONSTERS_TEAM=2;
}

